operator/main.go

```diff
@@ -492,7 +492,7 @@ func onOperatorStartLeading(ctx context.Context) {
 		}
 
 		if operatorOption.Config.SyncK8sNodes {
```

Comment:
I suppose you could add `if operatorOption.Config.KVStore != ""` right here... I must say that I think it would be nice to make use polymorphism, but not sure if adding a little bit of polymorphism here helps anything vs doing a some major refactoring.

Reply:
We can't exactly add that here in this block.
Since this is inside 
https://github.com/cilium/cilium/blob/b667f010fb814df3e056f18c39933e02e4b93bfb/operator/main.go#L418

Reply:
Ah, of course... I suppose @aanm would know best, but to me it looks that block might need refactoring.